homeassistant/components/proxmoxve/__init__.py

```diff
@@ -0,0 +1,153 @@
+                        vol.Required(CONF_USERNAME): cv.string,
+                        vol.Required(CONF_PASSWORD): cv.string,
+                        vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
+                        vol.Optional(CONF_REALM, default=DEFAULT_REALM): cv.string,
+                        vol.Optional(
+                            CONF_VERIFY_SSL, default=DEFAULT_VERIFY_SSL
+                        ): cv.boolean,
+                        vol.Required(CONF_NODES): vol.All(
+                            cv.ensure_list,
+                            [
+                                vol.Schema(
+                                    {
+                                        vol.Required(CONF_NODE): cv.string,
+                                        vol.Optional(CONF_VMS, default=[]): [
+                                            cv.positive_int
+                                        ],
+                                        vol.Optional(CONF_CONTAINERS, default=[]): [
+                                            cv.positive_int
+                                        ],
+                                    }
+                                )
+                            ],
+                        ),
+                    }
+                )
+            ],
+        )
+    },
+    extra=vol.ALLOW_EXTRA,
+)
+
+
+def setup(hass, config):
+    """Set up the component."""
+
+    # Create API Clients for later use
+    for entry in config[DOMAIN]:
+        host = entry[CONF_HOST]
+        port = entry[CONF_PORT]
+        user = entry[CONF_USERNAME]
+        realm = entry[CONF_REALM]
+        password = entry[CONF_PASSWORD]
+        verify_ssl = entry[CONF_VERIFY_SSL]
+
+        # Construct an API client with the given data for the given host
+        proxmox_client = None
+
+        try:
+            proxmox_client = ProxmoxClient(
+                host, port, user, realm, password, verify_ssl
+            )
+        except AuthenticationError:
+            _LOGGER.warning("Invalid credentials")
+
+        if proxmox_client is None:
+            _LOGGER.warning("Failed to connect to proxmox")
+            return False
+
+        PROXMOX_CLIENTS[f"{host}:{str(port)}"] = proxmox_client
+
+    hass.helpers.discovery.load_platform(
+        "binary_sensor", DOMAIN, {"entries": config[DOMAIN]}, config
+    )
+
+    return True
+
+
+class ProxmoxItemType(Enum):
+    """Represents the different types of machines in Proxmox."""
+
+    qemu = 0
+    lxc = 1
+
+
+class ProxmoxClient:
+    """A wrapper for the proxmoxer ProxmoxAPI client."""
+
+    def __init__(self, host, port, user, realm, password, verify_ssl):
+        """Initialize the ProxmoxClient."""
+
+        self._host = host
+        self._port = port
+        self._user = user
+        self._realm = realm
+        self._password = password
+        self._verify_ssl = verify_ssl
+
+        self.build_client()
```

Comment:
We don't want side effects in init method. Please call this method after creating the instance.